client/api/filesystem/v1alpha1/api.proto

```diff
@@ -2,27 +2,156 @@ syntax = "proto3";
 
 package v1alpha1;
 
-import "github.com/kubernetes-csi/csi-proxy/client/api/errors.proto";
-
-// FIXME: this is just an almost empty service, not actually implemented; it's just
-// here as an example of what API group and versions will look like. It will actually
-// be implemented in a later patch.
-
 service Filesystem {
-    // PathExists checks if the given path exists on the host.
+    // PathExists checks if the requested path exists in the host's filesystem
     rpc PathExists(PathExistsRequest) returns (PathExistsResponse) {}
+
+    // Mkdir creates a directory at the requested path in the host's filesystem
+    rpc Mkdir(MkdirRequest) returns (MkdirResponse) {}
+
+    // Rmdir removes the directory at the requested path in the host's filesystem.
+    // This may be used for unlinking a symlink created through LinkPath
+    rpc Rmdir(RmdirRequest) returns (RmdirResponse) {}
+
+    // LinkPath creates a local directory symbolic link between a source path 
+    // and target path in the host's filesystem
+    rpc LinkPath(LinkPathRequest) returns (LinkPathResponse) {}
+}
+
+// Context of the paths used for path prefix validation
+enum PathContext {
```

Comment:
Wouldn't we need a NONE type to represent absolute paths which are independent of the kubelet-csi-plugins-path/kubelet-pod-path ?

Reply:
We will not allow arbitrary absolute paths as was [discussed](https://github.com/kubernetes/enhancements/pull/1184#discussion_r308935813) during KEP review. May be I will add a note here clarifying that.